erts/emulator/beam/jit/arm/beam_asm.hpp

```diff
@@ -1532,6 +1532,19 @@ class BeamModuleAssembler : public BeamAssembler,
         flush_var(to2);
     }
 
+    void flush_vars(const Variable<arm::Gp> &to1,
+                    const Variable<arm::Gp> &to2,
+                    const Variable<arm::Gp> &to3) {
+        if (!to1.mem.hasBaseReg()) {
+            flush_vars(to2, to3);
+        } else if (!to2.mem.hasBaseReg()) {
+            flush_vars(to1, to3);
+        } else {
+            flush_vars(to1, to2);
+            flush_var(to3);
```

Comment:
What if `to1` (or `to2`) and `to3` are adjacent, but `to1` and `to2` are not? Should we check that?

Reply:
I will rewrite this to check for adjacent memory addresses.